File: patterns/java/src/refactoring_guru/patterns/factory_method/example/factory/WindowsDialog.java
package refactoring_guru.patterns.factory_method.example.factory;

import refactoring_guru.patterns.factory_method.example.buttons.Button;
import refactoring_guru.patterns.factory_method.example.buttons.WindowsButton;

/**
 * Windows Dialog will produce Windows buttons.
 */
public class WindowsDialog extends Dialog {

    @Override
    public Button createButton() {
        return new WindowsButton();
    }
}

File: patterns/cs/src/RefactoringGuru/FactoryMethod/Example/Factory/WindowsDialog.cs
using RefactoringGuru.FactoryMethod.Example.Buttons;

namespace RefactoringGuru.FactoryMethod.Example.Factory
{
    /**
     * EN: Windows Dialog will produce Windows buttons.
     * 
     * RU: Диалог на элементах операционной системы.
     */
    class WindowsDialog : Dialog
    {
        public override IButton CreateButton()
        {
            return new WindowsButton();
        }
    }
}
